backend/src/user-queries/query-parsing/parse-filter.ts

```diff
@@ -1 +1,104 @@
-// TODO filter step is not yet supported
+import {
+  FilterStep,
+  InferredSchemaColumnSchema,
+  OperatorsEnum,
+  Table,
+} from "@/definitions";
+import { compileTemplate } from "./utils";
+import * as _ from "lodash";
+import * as assert from "assert";
+
+const filterTemplate = `
+let {{ stepName }} = (
+  from {{ from }}
+  filter ({{ conditions }})
+)`;
+
+export const parseFilter = (
+  filterStep: FilterStep,
+  index: number,
+  tables: Table[],
+): string => {
+  // TODO move this into definitions
+  const logicalOperatorMap = {
+    and: "&&",
+    or: "||",
+  };
+
+  // Pull out logical operator for use in PRQL
+  if (!filterStep.logicalOperator) filterStep.logicalOperator = "and";
```

Comment:
Should we enforce this here or in FE + validation? I.e. not allowing a filter step without a logical op

Reply:
We currently do it in both places